pkg/core/statesync/module.go

```diff
@@ -344,7 +344,7 @@ func (s *Module) defineSyncStage() error {
 				s.syncStage |= mptSynced
```

Comment:
This check:
```
_, err = s.stateMod.GetStateRoot(s.syncPoint)
```

Is not reliable criteria of MPT sync:
1. Sync MPT
2. Interrupt node
3. Resume sync

State module doesn't have up-to-date root info (it's updated in a statejump callback). But MPT is in sync.

What we need is: add boolean field to Checkpooint denoting whether MPT was synced. Then here we need to retrieve checkpoint instead of statemodule's root.